scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerPage.java

```diff
@@ -436,77 +413,47 @@ public void onClick(MenuItemClickEvent event) {
-                SchedulerPage.this.controller.stopScheduler();
-            }
-        });
+        schedStopButton.addClickHandler(event -> SchedulerPage.this.controller.stopScheduler());
 
         schedFreezeButton = new MenuItem(FREEZE);
         schedFreezeButton.setIcon(SchedulerImages.instance.scheduler_freeze_16().getSafeUri().asString());
-        schedFreezeButton.addClickHandler(new com.smartgwt.client.widgets.menu.events.ClickHandler() {
-            public void onClick(MenuItemClickEvent event) {
-                SchedulerPage.this.controller.freezeScheduler();
-            }
-        });
+        schedFreezeButton.addClickHandler(event -> SchedulerPage.this.controller.freezeScheduler());
 
         schedResumeButton = new MenuItem(RESUME);
         schedResumeButton.setIcon(SchedulerImages.instance.scheduler_resume_16().getSafeUri().asString());
-        schedResumeButton.addClickHandler(new com.smartgwt.client.widgets.menu.events.ClickHandler() {
-            public void onClick(MenuItemClickEvent event) {
-                SchedulerPage.this.controller.resumeScheduler();
-            }
-        });
+        schedResumeButton.addClickHandler(event -> SchedulerPage.this.controller.resumeScheduler());
 
         schedPauseButton = new MenuItem(PAUSE);
         schedPauseButton.setIcon(SchedulerImages.instance.scheduler_pause_16().getSafeUri().asString());
-        schedPauseButton.addClickHandler(new com.smartgwt.client.widgets.menu.events.ClickHandler() {
-            public void onClick(MenuItemClickEvent event) {
-                SchedulerPage.this.controller.pauseScheduler();
-
-            }
-        });
+        schedPauseButton.addClickHandler(event -> SchedulerPage.this.controller.pauseScheduler());
 
         schedKillButton = new MenuItem(KILL);
         schedKillButton.setIcon(SchedulerImages.instance.scheduler_kill_16().getSafeUri().asString());
-        schedKillButton.addClickHandler(new com.smartgwt.client.widgets.menu.events.ClickHandler() {
-            public void onClick(MenuItemClickEvent event) {
-                SC.confirm("Do you really want to <strong>kill</strong> the Scheduler?", new BooleanCallback() {
-                    public void execute(Boolean value) {
-                        if (value)
-                            SchedulerPage.this.controller.killScheduler();
-                    }
-                });
-            }
-        });
+        schedKillButton.addClickHandler(event -> SC.confirm("Do you really want to <strong>kill</strong> the Scheduler?",
+                                                            value -> {
+                                                                if (value)
+                                                                    SchedulerPage.this.controller.killScheduler();
+                                                            }));
 
         ToolStripMenuButton adminMenuButton = new ToolStripMenuButton("Admin");
         this.adminMenu = new Menu();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)